opensearch-observability/src/main/kotlin/org/opensearch/observability/index/ObservabilityIndex.kt

```diff
@@ -0,0 +1,372 @@
+                } else {
+                    throw IllegalStateException("$LOG_PREFIX:Index $INDEX_NAME creation not Acknowledged")
+                }
+            } catch (exception: Exception) {
+                if (exception !is ResourceAlreadyExistsException && exception.cause !is ResourceAlreadyExistsException) {
+                    throw exception
+                }
+            }
+        }
+        if (isIndexExists(NOTEBOOKS_INDEX_NAME)) {
+            try {
+                log.info("$LOG_PREFIX:Index - reindex $NOTEBOOKS_INDEX_NAME to $INDEX_NAME")
+                val reindexResponse = ReindexRequestBuilder(client, ReindexAction.INSTANCE)
+                    .source(NOTEBOOKS_INDEX_NAME)
+                    .destination(INDEX_NAME)
+                    .abortOnVersionConflict(false)
+                    .refresh(true)
+                    .get()
+                if (reindexResponse.isTimedOut) {
+                    throw IllegalStateException("$LOG_PREFIX:Index - reindex $NOTEBOOKS_INDEX_NAME timed out")
+                } else if (reindexResponse.searchFailures.isNotEmpty()) {
+                    throw IllegalStateException("$LOG_PREFIX:Index - reindex $NOTEBOOKS_INDEX_NAME failed with searchFailures")
+                } else if (reindexResponse.bulkFailures.isNotEmpty()) {
+                    throw IllegalStateException("$LOG_PREFIX:Index - reindex $NOTEBOOKS_INDEX_NAME failed with bulkFailures")
+                }
+
+                log.info("$LOG_PREFIX:Index - ${reindexResponse.total} docs reindexed to $INDEX_NAME")
+                val deleteIndexRequest = DeleteIndexRequest(NOTEBOOKS_INDEX_NAME)
+                val actionFuture = client.admin().indices().delete(deleteIndexRequest)
+                val deleteIndexResponse = actionFuture.actionGet(PluginSettings.operationTimeoutMs)
+                if (deleteIndexResponse.isAcknowledged) {
+                    log.info("$LOG_PREFIX:Index $INDEX_NAME deletion Acknowledged")
+                } else {
+                    throw IllegalStateException("$LOG_PREFIX:Index $NOTEBOOKS_INDEX_NAME deletion not Acknowledged")
+                }
+            } catch (exception: Exception) {
+                if (exception !is ResourceNotFoundException && exception.cause !is ResourceNotFoundException) {
+                    throw exception
+                }
+            }
+        }
+    }
+
+    /**
+     * Check if the index is created and available.
+     * @param index
+     * @return true if index is available, false otherwise
+     */
+    private fun isIndexExists(index: String): Boolean {
+        val clusterState = clusterService.state()
+        return clusterState.routingTable.hasIndex(index)
+    }
+
+    /**
+     * Check if the index is created and available.
+     * @return true if index is available, false otherwise
+     */
+    private fun isIndexExists(): Boolean {
+        return isIndexExists(INDEX_NAME)
+    }
```

Comment:
Do we really need this function overloading?

Reply:
Will remove, yes it's unnecessary